x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/boosts/ProximityBoost.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.relevance.boosts;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.time.DateFormatter;
+import org.elasticsearch.common.time.DateMathParser;
+import org.elasticsearch.geometry.utils.GeographyValidator;
+import org.elasticsearch.geometry.utils.Geohash;
+import org.elasticsearch.geometry.utils.GeometryValidator;
+import org.elasticsearch.geometry.utils.WellKnownText;
+
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+import java.util.Objects;
+
+public class ProximityBoost extends ScriptScoreBoost {
+    private final GeometryValidator geoValidator = GeographyValidator.instance(true);
+    private final DateMathParser dateParser = DateFormatter.forPattern("date_optional_time||epoch_millis").toDateMathParser();
+
+    private final String center;
+    private final String function;
+    private final Float factor;
+
+    public static final String TYPE = "proximity";
+
+    public enum CenterType {
+        Numeric,
+        Geo,
+        Date
+    }
+
+    private CenterType centerType = null;
+
+    public ProximityBoost(String center, String function, Float factor) {
+        super(TYPE, "add");
+        this.center = center;
+        this.function = function;
+        this.factor = factor;
+        this.setCenterType();
+    }
+
+    public void setCenterType() {
+        centerType = null;
+        if (isNumber()) {
+            centerType = CenterType.Numeric;
+            return;
+        }
+        if (isDate()) {
+            centerType = CenterType.Date;
+            return;
+        }
+        if (isGeo()) {
+            centerType = CenterType.Geo;
+        }
+    }
+
+    private NumberFormat getNumberFormat() {
+        return NumberFormat.getInstance(Locale.ROOT);
+    }
+
+    public boolean isGeo() {
+        try {
+            String[] latLon = center.split(",", 2);
+            getNumberFormat().parse(latLon[0].trim());
+            getNumberFormat().parse(latLon[1].trim());
+            return true;
+        } catch (Exception e) {
+            // do nothing
```

Comment:
Do we at least want to log if we throw Exceptions in this class?

Reply:
Not really, not in this case. We're trying to parse and guess if the value is a geopoint, numeric value, or a datetime. It's expected that some of the parsing calls will fail. Why create more clutter in the logs.

Reply:
In that case, perhaps it would be better to streamline the Exception you're catching, for example catch `NumberFormatException`.